lib/message.ts

```diff
@@ -0,0 +1,89 @@
+import Discord from 'discord.js';
+
+import dearViktor from '../data/dearviktor.json';
+import commands from '../data/commands.json';
+import reactions from '../data/reactions.json';
+
+import { getKeyword } from './helpers';
+import { 
+    chooseRandom,
+    happensWithAChanceOf,
+} from './rng';
+import { Command } from '../lib/commands/commands'
+
+import { 
+    IReaction, 
+    IReactionDetails 
+} from './types/reaction';
+import { IDVKeywords } from './types/dearviktor';
+
+// LOGIC
+
+const isChannelDM = (msg:Discord.Message) => msg.author.id === msg.channel.id;
+const isUserAdmin = (msg:Discord.Message) => msg.member.hasPermission('ADMINISTRATOR');
+const isUserBot = (msg:Discord.Message) => msg.author.bot;
+const isUserArcy = (msg:Discord.Message) => msg.author.id === '165962236009906176';
+const messageStartsWithCommandSymbol = (msg:Discord.Message) => msg.content.startsWith(commands.options.commandSymbol);
+const isMessageRant = (msg:Discord.Message) => msg.content === msg.content.toUpperCase() && msg.content.length > 20;
+const isMessageDearViktor = (msg:Discord.Message) => msg.content.toLowerCase().startsWith('dear viktor');
+const isMessageDearVictor = (msg:Discord.Message) => msg.content.toLowerCase().startsWith('dear victor');
+
+const answer = (msg:Discord.Message, answer:string) => msg.channel.send(answer);
+const answerDearViktor = (msg:Discord.Message) => {
+    if (msg.content.endsWith('?')) {
+        const keywordDetected = dearViktor.keywords.find(
+            (category:any) => category.list.find(
+                (keyword:string) => msg.content.toLowerCase().indexOf(keyword) !== -1)
+            );
+        keywordDetected
+            ? answer(msg, chooseRandom(dearViktor.answers[keywordDetected.id]))
+            : answer(msg, chooseRandom(dearViktor.answers.yesno))
+        return;
+    }
+    answer(msg, '_That_ doesn\'t look like question to me.')
+};
+const answerDearVictor = (msg:Discord.Message) => answer(msg, '...what have you just call me. <:SilentlyJudging:288396957922361344>');
+const answerCommand = (msg:Discord.Message) => {
+    const command = commandObject(msg);
+    command
+        ? Command[getKeyword(msg)](command, msg)
```

Comment:
Currently if there's a command in the `command.json` file which is not declared in the `commands.js` file, this will crash. Needs some kind of proofchecking.